src/graph/SetExecutor.cpp

```diff
@@ -114,18 +108,41 @@ void SetExecutor::execute() {
     }
 
     auto *runner = ectx()->rctx()->runner();
-    runner->add([this] () mutable { left_->execute(); });
-    runner->add([this] () mutable { right_->execute(); });
 
-    auto cb = [this] (auto &&result) {
-        UNUSED(result);
-        if (!leftS_.ok() || !rightS_.ok()) {
+    auto lTask = [this] () mutable { left_->execute(); };
+    auto lError = [this] (auto &&e) {
+        std::stringstream ss;
+        ss << "Exception when handle left subquery: " << e.what();
+        LOG(ERROR) << ss.str();
+        leftS_ = Status::Error(ss.str());
+    };
+    auto lF = std::move(folly::makeFuture()).via(runner).then(lTask).thenError(lError);
```

Comment:
If `makeFuture` returns a future by value, the `std::move` is not needed. Am I right?